clang-tools-extra/docs/clang-tidy/checks/readability/use-builtin-literals.rst

```diff
@@ -0,0 +1,36 @@
+.. title:: clang-tidy - readability-use-builtin-literals
+
+readability-use-builtin-literals
+================================
+
+Finds literals explicitly casted to a type that could be expressed using builtin prefixes or suffixes.
+
+.. code-block: c++
+
+    (char)'a';                            // -> 'a'
+    (char16_t)U'a';                       // -> u'a'
+    static_cast<unsigned int>(0x1ul);     // -> 0x1u
+    reinterpret_cast<long long int>(3ll); // -> 3LL
+    float(2.);                            // -> 2.f
+    double{2.};                           // -> 2.
+
+Defined for character, integer and floating literals. Removes any suffixes or prefixes before applying the one that corresponds to the type of the cast. Long values will always have uppercase ``L`` recommended, conforming to ``readability-uppercase-literal-suffix``.
```

Comment:
If you do the following, the documentation page will create a clickable link to the other check's documentation. :wink: See https://clang.llvm.org/extra/clang-tidy/checks/cert/dcl16-c.html.
```suggestion
Defined for character, integer and floating literals. Removes any suffixes or prefixes before applying the one that corresponds to the type of the cast. Long values will always have uppercase ``L`` recommended, conforming to :doc:`readability-uppercase-literal-suffix <../readability/uppercase-literal-suffix>`.
```